### PR TITLE
Allow load server config from file and env vars

### PR DESCRIPTION
**Pull request by gnieto**
Use mehcode/config crate to configure the sources and priorites for
configuration loader.

I've moved all structs on `server/mod.rs` to a config module and I've
splitted in two distinct structures:

- RawConfig (I couldn't find a better name) which holds the
  configuration that can be changed by the users (via a TOML file or env
vars)
- Config: The calculated and parsed version, which holds all the
  materialized configuration.

As `CONFIG` is a static value and is used as a singleton on some places,
I've needed to add the `MAELSTROM_CONFIG_FILE` env var to allow loading
from a distinct file on tests.

I wanted to ask some questions:
- Are any concerns on depending on `mehcode/config` and/or `anyhow`?
- Which config format is prefered?  TOML? YAML? dotenv?
- Ia there any problem with prefix all env vars with `MAELSTORM`?

### CODE REVIEW

**Comment by chrisabruce** [comment]
No concerns with that Config crate or Anyhow as of now.
We should use yaml.  That is the default for other Home Server admins, so best to be consistent.
As for prefix, probably ok.  Maybe worth looking at what Synapse and Dendrite do, but we can always adjust later.

File: Cargo.toml [hunk 1]
```diff
@@ -10,12 +10,14 @@ actix-cors = "0.2.0"
 actix-ratelimit = "0.2"
 actix-rt = "1.0"
 actix-web = "2.0"
+anyhow = "1.0"
 async-trait = "0.1.30"
-dotenv = "0.15"
+config = { version = "0.10", default-features = false, features = [ "toml" ] }
```
Comment (chrisabruce) [inline Cargo.toml:15]: Can we default to `yaml`?

Comment (SuperSandro2000) [reply Cargo.toml]: Yaml is just so much nicer compared to toml.

**Comment by gnieto** [comment]
Changed to yaml

**Pull request merged**